pyflakes/checker.py

```diff
@@ -870,6 +895,33 @@ def ignore(self, node):
     # additional node types
     COMPREHENSION = KEYWORD = FORMATTEDVALUE = handleChildren
 
+    def DICT(self, node):
+        keys = node.keys
+        keys = [self.convert_to_value(key) for key in keys]
+        for key in set(keys):
+            if keys.count(key) > 1:
+                if isinstance(key, VariableKey):
+                    self.report(messages.DuplicateVariableDictionaryKey,
+                                node, key.name)
+                else:
+                    self.report(messages.DuplicateDictionaryKey, node, key)
+        self.handleChildren(node)
+
+    def convert_to_value(self, item):
+        if isinstance(item, ast.Str):
+            return item.s
+        elif isinstance(item, ast.Tuple):
+            return tuple(self.convert_to_value(i) for i in item.elts)
+        elif isinstance(item, ast.Num):
+            return item.n
+        elif isinstance(item, ast.Name):
```

Comment:
Im pretty sure this fails with instance properties and other attributes of a name.
Could you add it to the test suite , and discard names if I am correct about them.